extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxBlockingOutput.java

```diff
@@ -18,7 +18,8 @@
     protected boolean drainHandlerRegistered;
     protected final HttpServerRequest request;
     protected boolean first = true;
-    protected Throwable throwable;
+    protected volatile Throwable throwable;
+    protected ByteBuf data;
```

Comment:
Why are you now storing this ByteBuf?

Reply:
I retained a reference to it before calling `awaitWriteable()` [1], so that is if the connection is closed before the data has been written from the buffer, we can reset the buffer. [2]  If the buffer is pooled, my assumption was a call to write would end with an empty buffer.

The ref is nulled in the finally{} block in `write()` [3]

1 - https://github.com/quarkusio/quarkus/blob/af325984a82db5560cc320f2d5d1b659ee10e5ba/extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxBlockingOutput.java#L90

2 - https://github.com/quarkusio/quarkus/blob/af325984a82db5560cc320f2d5d1b659ee10e5ba/extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxBlockingOutput.java#L118

3 - https://github.com/quarkusio/quarkus/blob/af325984a82db5560cc320f2d5d1b659ee10e5ba/extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxBlockingOutput.java#L102